Client/Frontend/Home/HistoryHighlights/HistoryHighlightsCell.swift

```diff
@@ -0,0 +1,167 @@
+
+private struct RecentlyVisitedCellUX {
+    static let generalCornerRadius: CGFloat = 10
+    static let heroImageDimension: CGFloat = 24
+}
+
+struct RecentlyVisitedCellOptions {
+    let title: String
+    let heroImage: UIImage?
+    let corners: UIRectCorner?
+    let hideBottomLine: Bool
+    let isFillerCell: Bool
+
+    init(title: String,
+         shouldHideBottomLine: Bool,
+         with corners: UIRectCorner? = nil,
+         and heroImage: UIImage? = nil,
+         andIsFillerCell: Bool) {
+
+        self.title = title
+        self.hideBottomLine = shouldHideBottomLine
+        self.corners = corners
+        self.heroImage = heroImage
+        self.isFillerCell = andIsFillerCell
+    }
+
+    init(shouldHideBottomLine: Bool,
+         with corners: UIRectCorner? = nil,
+         andIsFillerCell: Bool) {
+
+        self.init(title: "",
+                  shouldHideBottomLine: shouldHideBottomLine,
+                  with: corners,
+                  and: nil,
+                  andIsFillerCell: andIsFillerCell)
+    }
+}
+
+/// A cell used in FxHomeScreen's History Highlights section.
+class HistoryHighlightsCell: UICollectionViewCell, ReusableCell {
+
+    // MARK: - UI Elements
+    let heroImage: UIImageView = .build { imageView in
+        imageView.contentMode = .scaleAspectFit
+        imageView.clipsToBounds = true
+        imageView.layer.masksToBounds = true
+        imageView.layer.cornerRadius = RecentlyVisitedCellUX.generalCornerRadius
+        imageView.image = UIImage.templateImageNamed("recently_closed")
+    }
+
+    let itemTitle: UILabel = .build { label in
+        // Limiting max size to accomodate for non-self-sizing parent cell.
+        label.font = DynamicFontHelper.defaultHelper.preferredFont(withTextStyle: .body,
+                                                                   maxSize: 23)
+        label.adjustsFontForContentSizeCategory = true
+    }
+
+    let itemDescription: UILabel = .build { label in
+        // Limiting max size to accomodate for non-self-sizing parent cell.
+        label.font = DynamicFontHelper.defaultHelper.preferredFont(withTextStyle: .caption1,
+                                                                   maxSize: 18)
+        label.adjustsFontForContentSizeCategory = true
+    }
+
+    private lazy var textStack: UIStackView = {
+        let stack = UIStackView(arrangedSubviews: [itemTitle, itemDescription])
+        stack.translatesAutoresizingMaskIntoConstraints = false
+        stack.distribution = .fillProportionally
+        stack.axis = .vertical
+        stack.alignment = .leading
+
+        return stack
+    }()
+
+    let bottomLine: UIView = .build { line in
+        line.isHidden = false
+    }
+
+    var isFillerCell: Bool = false {
+        didSet {
+            itemTitle.isHidden = isFillerCell
+            heroImage.isHidden = isFillerCell
+            bottomLine.isHidden = isFillerCell
+//            self.isUserInteractionEnabled = isFillerCell
```

Comment:
Can be removed? Or that's for the next PR?

Reply:
Next PR. :)